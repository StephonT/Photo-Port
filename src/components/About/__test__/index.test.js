import React from 'react';

//Jest creates a simulated DOM in a Node.js environment to approximate the DOM (no component is actually visibly rendered).
//The cleanup function is used to remove components from the DOM to prevent memory leaking, as well as variable or data collisions between tests that could corrupt test results.
import { render, cleanup } from '@testing-library/react';

import '@testing-library/jest-dom/extend-expect';

import About from '..';

//This will ensure that after each test, we won't have any leftover memory data that could give us false results.
afterEach(cleanup);

describe('About component', () => {
    //first test
    it('renders', ()=> {
        render(<About />);   
    });
    it('matches snapshot DOM node structure', () => {
     //asFragment function returns a snapshot of the About component   
        const { asFragment } = render(<About />); 
        expect(asFragment()).toMatchSnapshot();
    })
})