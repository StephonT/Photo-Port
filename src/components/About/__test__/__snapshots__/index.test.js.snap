// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <section
    class="my-5"
  >
    <h1
      id="about"
    >
      Who am I?
    </h1>
    <img
      alt="cover"
      class="my-2"
      src="cover-image.jpg"
      style="width: 100%;"
    />
  </section>
</DocumentFragment>
`;

//The index.test.js.snap file is a serialized version of the component's node structure, which includes elements, attributes, and text content. This file, stored in the __snapshots__ folder, will serve as the base model of the component's structure, and will be compared against new snapshots that are created by the asFragment function.